### ISSUE TITLE
[`imports-as-dependencies`] Node package not found

### ISSUE THREAD

**Issue by regseb** [gajus/eslint-plugin-jsdoc#1112]
## Expected behavior

[`imports-as-dependencies`](https://github.com/gajus/eslint-plugin-jsdoc/blob/main/docs/rules/imports-as-dependencies.md#readme) should support Node package (e.g. `http`, `path`).

## Actual behavior

`imports-as-dependencies` reports: _import points to package which is not found in dependencies_

### `package.json`

```json
{
  "name": "testcase",
  "dependencies": {
    "eslint": "8.42.0",
    "eslint-plugin-jsdoc": "46.2.4",
    "typescript": "5.1.3"
  },
  "eslintConfig": {
    "plugins": ["jsdoc"],
    "rules": {
      "jsdoc/imports-as-dependencies": "error"
    }
  }
}
```

### `tsconfig.json`

```json
{
  "compilerOptions": {
    "allowJs": true,
    "noEmit": true,
    "checkJs": true
  }
}
```

### `index.js`

```js
/**
 * @typedef {import("node:http").Server} Server
 */

/**
 * @param {Server} server
 */
module.exports = function foo(server) {
    console.log(server.maxHeadersCount);
};
```

## To reproduce

`npx eslint index.js`

```
/home/regseb/testcase/index.js
  2:1  error  import points to package which is not found in dependencies  jsdoc/imports-as-dependencies

✖ 1 problem (1 error, 0 warnings)
```

## Environment

- Node version: 18.16.0
- ESLint version: 8.42.0
- `eslint-plugin-jsdoc` version: 46.2.4

## Additional info

- `npx tsc` support Node package (no error reported).
- You can use `@typedef {import("http").Server} Server` (without `node:` prefix).

**Comment by brettz9** [comment]
Makes sense, but shouldn't `@types/node` be detected as present?

**Comment by regseb** [comment]
I don't have `@types/node` in the dependencies.

If I add it, there's always the error. I think the rule can't make the connection between `node:http` and the dependency `@types/node`.

You may need to add a list of allowed imports: `["node:assert", "node:async_hooks", "node:async_hooks", "node:buffer", /* ... */, "node:zlib"]`

**Comment by brettz9** [comment]
Yes, I've already prepared a PR. I'm just wondering whether the desired behavior should require `@types/node` or nothing at all.

**Comment by brettz9** [comment]
Since Intellisense works without `@types/node` installed, I guess `@types/node` is not strictly required.
